src/kademlia/dht_storage.cpp

```diff
@@ -532,6 +556,36 @@ namespace
 			if (!peers.empty() && peers.capacity() / peers.size() >= 4u)
 				peers.shrink_to_fit();
 		}
+
+		void refresh_infohashes_sample()
```

Comment:
isn't there a way to assign a ``span`` to an entry? Either way, I would consider this primarily a problem in ``entry``. It seems like it would be cleaner for this class to not worry _too_ much about how it's encoded. But the bytes in memory *would* be the same as with the string, right?